### DISCUSSION TITLE
Make response-ratelimiting plugin and rate-limiting plugin return the same response when limited (or at least be more consistent).

### DISCUSSION THREAD

**Discussion by nricklin** [Kong/kong#7211] [Ideas and feature requests]
When rate limited, the rate-limiting plugin returns a `429` with the following response body:

``` json
{"message":"API rate limit exceeded"}
```

However the response-ratelimiting plugin returns a `429` but with a blank response body.

I propose to make response-ratelimiting return the same message, I assume simply by tweaking this line to include a message:
https://github.com/Mashape/kong/blob/923639ba45253108a0a1940cd359bbb93f58128b/kong/plugins/response-ratelimiting/header_filter.lua#L54

I think it offers a more consistent experience for API users to see the same error message when rate limited.

I'll do a pull request if it makes sense to others.

**Comment by sonicaghi** [comment]
@thefosk ^

**Comment by thibaultcha** [comment]
Sounds good to me, the only exception being that the response-ratelimiting plugin is supposed to be renamed "quotas" as per #579, so the response should probably be:

``` json
{"message":"API quota exceeded"}
```

PRs welcome, indeed

**Comment by nricklin** [comment]
Alright, it's in there in pull request #733 with a response of:

``` json
{"message":"API quota exceeded"}
```

It was a bit more than that one-liner.  I had to move the blocking response from the header_filter to the action part of the plugin.

**Comment by nricklin** [comment]
I just verified in kong 0.5.2 (using the AWS AMI) that if you set a response-ratelimiting limit to `N`, then this will happen:
- The first N requests make it through correctly
- Request N+1 does in fact hit the upstream server, but the response from kong is 429.

This pull request #733 should fix this issue as well.
